difflint/lint.py

```diff
@@ -1,56 +1,152 @@
+            if shutil.which(linter) is None:
+                missing_linters.append({'language': language,
+                                        'linter': linter})
+    return missing_linters
 
 
-def _lint_javascript(file_to_lint):
-    output = LintOutput()
+def _lint_jscs(file_to_lint, lint_output):
     jscs = shutil.which('jscs')
-    output.run_command([jscs, file_to_lint, '--reporter',
-                        resource_filename(__name__,
-                                          'data/jscs_terse_reporter.js')])
+    lint_output.run_command([jscs, file_to_lint, '--reporter',
+                             resource_filename(__name__,
+                                               'data/jscs_terse_reporter.js')])
+    return lint_output
+
 
+def _lint_jshint(file_to_lint, lint_output):
     jshint = shutil.which('jshint')
-    output.run_command([jshint, file_to_lint, '--reporter',
-                        resource_filename(__name__,
-                                          'data/jshint_terse_reporter.js')])
-    return output
+    lint_output.run_command([jshint, file_to_lint, '--reporter',
+                             resource_filename(__name__,
+                                               'data/jshint_terse_reporter.js')])
+    return lint_output
 
 
-def _lint_python(file_to_lint):
-    output = PythonLintOutput()
-    output.lint_pep8(file_to_lint)
-    output.lint_pyflakes(file_to_lint)
-    return output
+def _lint_pep8(file_to_lint, lint_output):
+    reporter = PEP8TerseReporter()
+    checker = pep8.Checker(filename=file_to_lint, report=reporter)
+    num_problems = checker.check_all()
+    if num_problems > 0:
+        lint_output._warnings_present = True
+    lint_output.output += reporter.output
+    return lint_output
+
+
+def _lint_pyflakes(file_to_lint, lint_output):
+    reporter = PyFlakesTerseReporter()
+    num_problems = pyflakes.api.checkPath(file_to_lint, reporter=reporter)
+    if num_problems > 0:
+        lint_output._warnings_present = True
+    lint_output.output += reporter.output
+    return lint_output
 
 
 def lint(file_to_lint):
     """Perform linting on a file according to its extension.
 
-    Inputs:
-        file_to_lint: Path to a file to lint, as a string.
+    Inputs: Path to a file to lint, as a string.
+
     Output: A LintOutput object with linting results.
     """
     root, ext = os.path.splitext(file_to_lint)
 
-    if ext == '.js':
-        return _lint_javascript(file_to_lint)
-    if ext == '.py':
-        return _lint_python(file_to_lint)
+    # Strip the leading period to match the format in the configuration file.
+    ext = ext[1:]
+
+    config = _read_enabled_linters_config()
+
+    linters_to_run = []
+    for language in config:
+        if ext in config[language]['extensions']:
+            linters_to_run.extend(config[language]['linters'])
 
-    # TODO: Other file extensions/linters will be added here.
+    output = LintOutput()
```

Comment:
Perhaps, but this gives a path for a theoretical linter that affects two different (presumably related) languages to be used.